src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx

```diff
@@ -82,7 +84,13 @@ const EnvConfig = ({ filteredAppIds, envName }: AppGroupDetailDefaultType) => {
         }
     }, [initDataResults])
 
-    if (loading || !envAppList.length) {
+    useEffect(() => {
```

Comment:
Why is this useEffect required?

Reply:
If the app is unavailable in the current environment, redirect to the app selection page
added comment in code

Reply:
Can't we achieve the same without useEffect?

Reply:
discussed